arches/app/views/api/__init__.py

```diff
@@ -2383,3 +2383,128 @@ def delete(self, request, identifier=None):
+            overall_access = True
+
+        if identifier:
+            try:
+                returned_user_preference = models.UserPreference.objects.get(
+                    pk=identifier
+                )
+            except:
+                return JSONErrorResponse(
+                    _("User Preference GET request failed"),
+                    _("No User Preference with this id"),
+                    status=404,
+                )
+            if overall_access or returned_user_preference.user == request.user:
+                response_data = returned_user_preference
+            else:
+                return JSONErrorResponse(
+                    _("User Preference GET request failed"),
+                    _("You do not have access to view this userpreferenceid"),
+                    status=403,
+                )
+        else:
+            if overall_access:  # give view to everything
+                response_data = [x for x in models.UserPreference.objects.all()]
+            else:
+                response_data = [
+                    x for x in models.UserPreference.objects.filter(user=request.user)
+                ]
+
+        return JSONResponse(response_data)
+
+    @method_decorator(group_required("Application Administrator", raise_exception=True))
+    def post(self, request, identifier=None):
+        if identifier:
+            return JSONErrorResponse(
+                _("User Preference creation failed"),
+                _(
+                    "POST request should not have a userpreferenceid provided in the URL"
+                ),
+                status=400,
+            )
+
+        try:
+            user_pref_json = JSONDeserializer().deserialize(request.body)
+        except ValueError:
+            return JSONErrorResponse(
+                _("Invalid JSON data"),
+                _("The User Preference API was sent invalid JSON"),
+                status=400,
+            )
+
+        if user_pref_json:
+            if user_pref_json["userpreferenceid"]:
+                JSONErrorResponse(
+                    _("Incorrect User Preference json data"),
+                    _(
+                        "POST REST request should not have a userpreferenceid provided in the JSON data."
+                    ),
+                    status=400,
+                )
+            else:
+                try:
+                    models.User.objects.get(username=user_pref_json["user"])
+                except:
+                    return JSONErrorResponse(
+                        _("Invalid user"),
+                        _("The User Preference API includes an invalid user."),
+                        status=400,
+                    )
+                new_user_preference = models.UserPreference()
+                new_user_preference.userpreferenceid = uuid.uuid4()
+                new_user_preference.user = models.User.objects.get(
+                    username=user_pref_json["user"]
+                )
```

Comment:
Either just set `.user_id` or save the object you retrieved above to avoid re-querying.